### PR TITLE
Add Cassandra script

### PR DESCRIPTION
**Pull request by madhur**
Adds the script for [Cassandra](https://github.com/apache/cassandra)

## Description

I am a backend engineer and see good use cases of having backend systems in this repo. I observed MariaDB, Postgres are already there.
I foresee having tools like Cassandra, Redis, MongoDB would be of great use to backend engineers.

## Type of change

Please delete options that are not relevant.
- [x] New Script

### CODE REVIEW

**Comment by tteck** [comment]
Are you currently utilizing Apache Cassandra with Proxmox?

**Comment by madhur** [comment]
Yes. I run my local java services on laptop  for development purposes.

They require mysql, Cassandra, redis and Mongo which run on my Proxmox box.

**Comment by tteck** [comment]
Why didn't you directly modify `cassandra.yaml` in place instead of using `cat` to recreate it?

**Comment by madhur** [comment]
I can do that too. I saw the example of  [`blocky`](https://github.com/tteck/Proxmox/blob/main/install/blocky-install.sh) and just followed it.

**Comment by tteck** [comment]
The following changes need to be made before it would be considered.

```
msg_info "Installing Cassandra"
release=$(curl -s https://cassandra.apache.org/_/download.html | grep -oP '(?<=/cassandra/)[0-9]+\.[0-9]+\.[0-9]+' | head -n 1)

wget -qO- https://dlcdn.apache.org/cassandra/$release/apache-cassandra-${release}-bin.tar.gz | tar -xz -C /opt
mv /opt/apache-cassandra-${release} /opt/cassandra

sed -i -e 's/^rpc_address: localhost/#rpc_address: localhost/g' -e 's/^# rpc_interface: eth1/rpc_interface: eth0/g' /opt/cassandra/conf/cassandra.yaml
msg_ok "Installed Cassandra"
```

**Comment by madhur** [comment]
Understood. Let me test out these and ping here again.

**Comment by tteck** [comment]
Did it work for you?

**Comment by madhur** [comment]
Give me some more time to test. Just realized this line https://github.com/tteck/Proxmox/pull/2367/files#diff-affd204081724acf266d1e360d1adf21195ddbb2a74b205ca4d415d096620440R2 was left pointing to my branch.

I have to change line ` lxc-attach -n "$CTID" -- bash -c "$(wget -qLO - https://raw.githubusercontent.com/tteck/Proxmox/main/install/$var_install.sh)" || exit` in `build.func` to point to my branch.

Not sure if there is a better way to test?

**Comment by madhur** [comment]
Have tested out the changes. Looks good now.

**Comment by tteck** [comment]
Following its creation, how can one verify that it is functioning as intended?

**Comment by madhur** [comment]
One can connect to using cassandra client. For example using official cassandra client in `/bin/cqlsh`

```
 ./cqlsh 192.168.1.60                                                                                                                                                                                                                                                             
Connected to Test Cluster at 192.168.1.60:9042
```

File: install/cassandra-install.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+msg_info "Installing OpenJDK"
+wget -qO- https://packages.adoptium.net/artifactory/api/gpg/key/public | gpg --dearmor >/etc/apt/trusted.gpg.d/adoptium.gpg
+echo 'deb [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/adoptium.gpg] https://packages.adoptium.net/artifactory/deb bookworm main' >/etc/apt/sources.list.d/adoptium.list
+$STD apt-get update
+$STD apt-get install -y temurin-8-jdk
+msg_ok "Installed OpenJDK"
+
+msg_info "Installing Cassandra"
+cd /opt
```
Comment (tteck) [inline install/cassandra-install.sh]: Needs to be removed

Comment (madhur) [reply install/cassandra-install.sh]: Removed

**Comment by tteck** [comment]
Thanks

**Comment by madhur** [comment]
Thanks for the awesome scripts

**Comment by tteck** [comment]
I'm wondering if this would be a better install method. Easy updates.

```
wget -qO- https://downloads.apache.org/cassandra/KEYS | gpg --dearmor >/etc/apt/trusted.gpg.d/cassandra.gpg
echo "deb https://debian.cassandra.apache.org 41x main" >/etc/apt/sources.list.d/cassandra.sources.list
apt-get update
apt-get cassandra cassandra-tools
```
What are your thoughts?

**Comment by madhur** [comment]
Yes, Agreed. I wan't aware it is available in debian repos.

**Comment by madhur** [comment]
The changes to conf will still be required though in order to allow access from outside of container.

**Comment by tteck** [comment]
I think that this installation method is the way.

```
root@cassandra:~# cqlsh 192.168.86.253
Connected to Test Cluster at 192.168.86.253:9042
[cqlsh 6.1.0 | Cassandra 4.1.3 | CQL spec 3.4.6 | Native protocol v5]
Use HELP for help.
cqlsh> exit
root@cassandra:~# 
```

**Pull request merged**